src/dsverifier.cpp

```diff
@@ -1841,207 +1934,591 @@ void check_minimum_phase_delta_domain()
  Purpose:
 
  \*******************************************************************/
-
-void check_state_space_stability()
+int check_state_space_stability()
 {
-  Eigen::MatrixXd matrixA = Eigen::MatrixXd::Ones(_controller.nStates,
-      _controller.nStates);
+  Eigen::MatrixXd matrixA(_controller.nStates, _controller.nStates);
   int i, j;
-
   for(i = 0; i < _controller.nStates; i++)
   {
     for(j = 0; j < _controller.nStates; j++)
     {
-      // fxp_double_to_fxp(A[i][j]);
-      matrixA(i, j) = _controller.A[i][j];
+      matrixA(i, j) = static_cast<double>(_controller.A[i][j]);
     }
   }
-
   std::complex<double> lambda;
   std::complex<double> margem(1, 0);
-
+  double v;
   dsverifier_messaget dsv_msg;
+  Eigen::VectorXcd eivals = matrixA.eigenvalues();
   for(i = 0; i < _controller.nStates; i++)
   {
-    lambda = matrixA.eigenvalues()[i];
-    std::cout << "abs(lambda): " << std::abs(lambda) << std::endl;
-    double v = std::abs(lambda);
+    lambda = eivals[i];
+    v = std::sqrt(lambda.real()*lambda.real()+lambda.imag()*lambda.imag());
     if(v > 1.0)
     {
-      dsv_msg.show_verification_failed(); // unstable system
-      exit(0);
+      std::cout << "unstable: " << std::endl;
+      return 0; // unstable system
     }
   }
-
-  dsv_msg.show_verification_successful(); // stable system
+  return 1; // stable system
 }
 
 /*******************************************************************
- Function: generates_mag_response
+ Function: isEigPos
 
  Inputs:
 
  Outputs:
 
- Purpose:
+ Purpose: Check if two variables are both positive or both negative
 
  \*******************************************************************/
+bool isEigPos(Eigen::MatrixXd A)
+{
+  int isStable = check_state_space_stability();
```

Comment:
Shouldn't it be a boolean?

Reply:
That is because check_state_space_stability() function returns an integer.